lib/FundDistribution/FundDistributionFields/FundDistributionFields.js

```diff
@@ -22,50 +24,108 @@ import {
+    acc[distr.fundId] = calculateFundAmount(distr, totalAmount);
+
+    return acc;
+  }, {});
+  const remainingAmount = totalAmount - sum(Object.values(amounts));
+  const remainingAmountNode = (
+    <React.Fragment>
+      <FormattedMessage
+        id="stripes-acq-components.fundDistribution.remainingAmount"
+        values={{
+          remainingAmount: (
+            <AmountWithCurrencyField
+              currency={currency}
+              amount={remainingAmount}
+            />
+          ),
+        }}
+      />
+    </React.Fragment>
+  );
+
+  const onAdd = (fields) => {
+    fields.push({
+      distributionType: FUND_DISTR_TYPE.percent,
+    });
+  };
+
+  // eslint-disable-next-line react/prop-types
+  const renderTypeToggle = ({ input: { value, onChange } }) => {
+    return (
+      <KeyValue label={<FormattedMessage id="stripes-acq-components.fundDistribution.type" />}>
+        <ButtonGroup
+          fullWidth
+          data-test-fund-distr-type
+        >
+          <Button
+            onClick={() => onChange(FUND_DISTR_TYPE.percent)}
+            buttonStyle={value === FUND_DISTR_TYPE.percent ? 'primary' : 'default'}
+            data-test-fund-distr-type-percent
+            disabled={disabled}
+          >
+            <FormattedMessage id="stripes-acq-components.fundDistribution.type.sign.percent" />
+          </Button>
+          <Button
+            onClick={() => onChange(FUND_DISTR_TYPE.amount)}
+            buttonStyle={value === FUND_DISTR_TYPE.amount ? 'primary' : 'default'}
+            data-test-fund-distr-type-amount
+            disabled={disabled}
+          >
+            <FormattedMessage id="stripes-acq-components.fundDistribution.type.sign.amount" />
+          </Button>
+        </ButtonGroup>
+      </KeyValue>
+    );
+  };
+
   const renderSubForm = (elem, index, fields) => {
     const fund = fields.get(index);
-    const fundCode = get(funds.find(({ value }) => value === fund.fundId), 'code');
 
     return (
       <Row>
-        <Col xs={4}>
+        <Col xs>
           <FieldSelection
             dataOptions={funds}
             labelId="stripes-acq-components.fundDistribution.id"
             name={`${elem}.fundId`}
             required={required}
-            validate={required && [validateRequired]}
             disabled={disabled}
           />
         </Col>
-        <Col xs={2}>
+        <Col xs>
           <Field
             component={TextField}
             fullWidth
-            label={<FormattedMessage id="stripes-acq-components.fundDistribution.percent" />}
-            name={`${elem}.percentage`}
+            label={<FormattedMessage id="stripes-acq-components.fundDistribution.value" />}
+            name={`${elem}.value`}
             type="number"
             parse={parseNumberFieldValue}
             required={required}
-            validate={required && [validateRequired]}
+            validate={required ? [validateRequired] : undefined}
```

Comment:
this is to remove console warning (`false` is not appropriate value for `validate` prop)